sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_lease_client.hpp

```diff
@@ -55,7 +57,11 @@ namespace Azure { namespace Storage { namespace Blobs {
      *
      * @return Lease id of this lease client.
      */
-    const std::string& GetLeaseId() const { return m_leaseId; }
+    std::string GetLeaseId()
+    {
+      std::lock_guard<std::mutex> guard(m_mutex);
+      return m_leaseId;
```

Comment:
Is the lease id always the same size()?  or do we know enough about how it will come back from the server so we can use a trivially copyable component like a vector/array ?  Because in that case, we could use std::atomic instead of the mutex.

Reply:
Yeah, I'm sure this kind of optimization will be trivial.

Reply:
We shouldn't worry about this kind of stuff given the lease APIs are going to make network I/O.